tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/JCardObjectDeserializer.java

```diff
@@ -0,0 +1,72 @@
+package com.linagora.tmail.contact;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.james.util.streams.Iterators;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+
+public class JCardObjectDeserializer extends StdDeserializer<JCardObject> {
+    private static final String FN = "fn";
+    private static final String EMAIL = "email";
+    private static final Set<String> SUPPORTED_PROPERTY_NAMES = Set.of(FN, EMAIL);
+    private static final int PROPERTY_NAME_INDEX = 0;
+    private static final int PROPERTIES_ARRAY_INDEX = 1;
+    private static final int TEXT_PROPERTY_VALUE_INDEX = 3;
+
+    public JCardObjectDeserializer() {
+        this(null);
+    }
+
+    protected JCardObjectDeserializer(Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public JCardObject deserialize(JsonParser p, DeserializationContext ctxt)
+        throws IOException {
+        JsonNode node = p.getCodec().readTree(p);
+
+        JsonNode jCardPropertiesArray = node.get(PROPERTIES_ARRAY_INDEX);
+        Map<String, String> jCardProperties =
+            collectJCardProperties(jCardPropertiesArray.iterator());
+
+        if (!jCardProperties.containsKey(FN)) {
+            throw new RuntimeException("The FN field is required according to specification.");
+        }
```

Comment:
I believe we should be more lenient (be lenient on what you accept strict on what you send)

Can we make FN optional in our datmodel? Or position it to an empty string?